### DISCUSSION TITLE
`createPages`'s `shouldRunCreatePagesStatefully` option: Public API?

### DISCUSSION THREAD

**Discussion by tony** [gatsbyjs/gatsby#35061] [Help] [answered]
Is this an internal API or a public one?

I have a 20,000+ page site, working on build performance, checking innards of `createPage`.

My guess is it's an internal API.

https://github.com/gatsbyjs/gatsby/blob/f7fb383d78319f1b81272896208004362146174f/packages/gatsby/src/services/create-pages.ts#L10-L17

https://github.com/gatsbyjs/gatsby/blob/f7fb383d78319f1b81272896208004362146174f/packages/gatsby/src/services/create-pages.ts#L60-L79

https://github.com/gatsbyjs/gatsby/search?q=shouldRunCreatePagesStatefully&type=code

via #28590 a0b31bcaaa04db8ff0675b3653c841bf47c67814

I'm relatively new to profiling build systems like this

**Comment by KyleAMathews** [comment, marked as answer]
It is a private api yes. 

Curious why you're looking here? A 20k page site should take 4-6s to create pages and isn't generally the slowest part of a build.

**Reply by tony** [reply]
Correct and alas, it's not createPages

```
success createPages - 5.593s
success createPagesStatefully - 0.084s
info Total nodes: 39114, SitePage nodes: 29956 (use --verbose for breakdown)
```

Your assessment about the createPages seems right.

I'm proceeding to later in the build phases identify where the bottleneck is.

P.S. `gatsby-plugin-schema-snapshot` has given a significant boost when using `gatsby start`
